src/test/java/seedu/address/model/tag/TagTest.java

```diff
@@ -1,20 +1,30 @@
 package seedu.address.model.tag;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static seedu.address.testutil.Assert.assertThrows;
 
 import org.junit.jupiter.api.Test;
 
 public class TagTest {
+    @Test
+    public void factory_sameValue_sameInstance() {
+        // same string returns same object
+        boolean equal = Tag.of("tag") == Tag.of("tag");
+        assertEquals(true, equal);
+
+        // null
+        assertEquals(null, Tag.of(null));
+    }
 
     @Test
-    public void constructor_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> new Tag(null));
+    public void factory_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> Tag.of(null));
```

Comment:
The gradle test showed this test has failed?

<img width="222" alt="image" src="https://github.com/AY2324S1-CS2103T-F12-4/tp/assets/107026005/f069d186-9151-432b-ba17-c0acf4a8ccbe">

Reply:
oops i forgot to test again after changing the method